pkgs/development/python-modules/jaxlib/default.nix

```diff
@@ -81,7 +81,7 @@ let
   cudatoolkit_cc_joined = symlinkJoin {
     name = "${cudatoolkit.cc.name}-merged";
     paths = [
-      cudatoolkit.cc
+      backendStdenv.cc
       binutils.bintools # for ar, dwp, nm, objcopy, objdump, strip
     ];
   };
```

Comment:
Jaxlib's Bazel BUILD ignores the stdenv, and it also ignores particular the libstdc++ it propagates as build inputs. In particular, their BUILD needs `ar` and `cc` to be located in the same directory. I remember that overriding `buildBazelPackage` for tensorflow did actually work, and I would guess the reason we don't do that for jaxlib is because we tried and it didn't work out. I'll give another try though, thanks!

Reply:
Random guess: It could be that Bazel would ignores the `AR` and `CC` environment variables because the jax cuda config uses a custom `cc_crosstool_top`. I believe Jax for CUDA pulls the `@local_config_cuda` from the XLA repo, probably from [here](https://github.com/openxla/xla/tree/main/third_party/gpus). It might be that some cc_toolchain_config there overrides or deletes the AR and CC values.

Reply:
> Do both the regular stdenv and backendStdenv need to be present when building with CUDA support?

No, they don't, we could as well override `buildBazelPackage`. Just that otherwise it would be a no-op

Reply:
> I would guess the reason we don't do that for jaxlib is because we tried and it didn't work out.

I'm not aware of any attempts using CMake for jaxlib, but it's possible that someone tried it and never submitted a PR. I'm hesitant to veer too far off upstream's build system, but upstream might actually be sympathetic to a CMake build as well.